src/sentry/rules/actions/base.py

```diff
@@ -229,20 +267,21 @@ def get_dynamic_form_fields(self):
             return fields
         return form_fields
 
-    def translate_integration(self, integration):
-        return integration.name
+    def translate_integration(self, integration: Integration) -> str:
+        name: str = integration.name
+        return name
 
     @property
-    def prompt(self):
+    def prompt(self) -> str:
         return f"Create {self.ticket_type}"
 
-    def generate_footer(self, rule_url):
+    def generate_footer(self, rule_url: str) -> str:
         raise NotImplementedError
 
-    def after(self, event, state):
+    def after(self, event: Event, state: EventState) -> Generator[CallbackFuture, None, None]:
         integration_id = self.get_integration_id()
         key = f"{self.provider}:{integration_id}"
-        return self.future(
+        yield self.future(
```

Comment:
Are you sure you can switch `return` and `yield` here?

Reply:
No, I think this would be a bug.